airbyte-webapp/src/core/domain/connection/utils.ts

```diff
@@ -1,4 +1,18 @@
-import { WebBackendConnectionRead, WebBackendConnectionUpdate } from "core/request/AirbyteClient";
+import {
+  ConnectionScheduleData,
+  ConnectionScheduleType,
+  WebBackendConnectionRead,
+  WebBackendConnectionUpdate,
+} from "core/request/AirbyteClient";
+
+interface ConnectionScheduleProps {
+  scheduleType?: ConnectionScheduleType | null;
+  scheduleData?: ConnectionScheduleData | null;
+}
+
+export const getConnectionScheduleType = (connection: ConnectionScheduleProps): ConnectionScheduleType =>
+  connection.scheduleType ??
+  (connection.scheduleData?.basicSchedule ? ConnectionScheduleType.basic : ConnectionScheduleType.manual);
```

Comment:
I think this needs to be reversed. Right now if you go from manual, to a time, then back it has an error.

Reply:
I decided not to overcomplicate this and reverted the changes. The real problem is that scheduleType was missing when building the update. But the connection object always returns a `scheduleType`